File: js/problem1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    if (nums.length < 2) {
        return [];
    }
    
    if (nums.length == 2) {
        if (nums[0] + nums[1] === target) {
            return [0, 1];
        }
        return [];
    }
    
    let ht = new Map();
    ht.set(nums[0], 0);
    
    for (let i = 1; i < nums.length; i++) {
        let diff = target - nums[i];
        if (ht.has(diff)) {
            return [ht.get(diff), i];
        }
        ht.set(nums[i], i);
    }
    
    return [];
};

File: leetcodeJava/src/main/java/Problem1.java
// 1. Two Sum

class Solution {
    public int[] twoSum(int[] nums, int target) {
        if (nums.length < 2) {
            return new int[0];
        }
        
        Map<Integer, Integer> ht = new HashMap<Integer, Integer>();
        ht.put(nums[0], 0);
        for (int i = 1; i < nums.length; i++) {
            int diff = target - nums[i];
            if (ht.containsKey(diff)) {
                return new int[] {ht.get(diff), i};
            }
            ht.put(nums[i], i);
        }
        return new int[0];
    }
}